### ISSUE TITLE
journal-upload shouldn't quit if the connection to remote host fails

### ISSUE THREAD

**Issue by utezduyar** [systemd/systemd#2877]
### Submission type
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> …225

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> …Own distro
### Details:

systemd-journal-upload fails with return code 1 if the connection to remote host fails. There could always be a glitch or something unexpected on the remote side. To be able to have a retry mechanism, you need to have a new service. 

I am proposing to have a retry functionality in systemd-journal-upload if connection to the remote host fails.

**Comment by archuserr** [comment]
I agree. It still works this way in v232. systemd-timesyncd doesn't exit if it can't connect to the time server. It sleeps for 30 seconds and tries again. Why doesn't systemd-journal-upload work the same way?

**Comment by karme** [comment]
a simple solution would be to use Restart and RestartSec in the corresponding service file

**Comment by utdrmac** [comment]
Holy bejezus! This bug is 11+ years old?!? I just learned about systemd-journal-upload yesterday, and am attempting to implement it across our fleet, sending logs to a central victorialogs server. If the journald entry is too large, or basically any error whatsoever is encountered, systemd-journal-upload dies and never restarts. That's crazy in 2025. This is centralized logging; how can it NOT auto-retry?
